tgfloofbot/plugins/administration/commands.py

```diff
@@ -0,0 +1,117 @@
+
+    admin_group_id = client.config.main_group
+
+    def _resolve_admin_group_id(update):
+        try:
+            admin_group = client.updater.bot.get_chat(chat_id=admin_group_id)
+        except telegram.error.TelegramError:
+            LOG.critical("Unable to resolve the group ID set in the config! Defaulting to context group ID.")
+            admin_group = update.effective_chat
+
+        return admin_group
+
+    class WarnCommandArgs(pydantic.BaseModel):
+        bad_user: Optional[str] = pydantic.Field(
+            None, description="A raw user ID")
+        warn_message: Optional[str] = pydantic.Field(
+            None, description="The warning reason")
+
+    class UsernoteCommandArgs(pydantic.BaseModel):
+        bad_user: Optional[str] = pydantic.Field(
+            None, description="A raw user ID")
+        warn_message: Optional[str] = pydantic.Field(
+            None, description="The note")
+
+    @loader.command(name="warn", help="warn a user", admin=True)
+    def warn_command(
+        client: TGFloofbotClient,
+        update: Update,
+        context: CallbackContext,
+        args: WarnCommandArgs
+    ) -> None:
+        warn_helper(client, update, context, args, False)
+        return
+
+    @loader.command(name="note", help="adds a moderation note for a user", admin=True)
+    def usernote_command(
+        client: TGFloofbotClient,
+        update: Update,
+        context: CallbackContext,
+        args: UsernoteCommandArgs
+    ) -> None:
+        warn_helper(client, update, context, args, True)
+        return
+
+    def warn_helper(client, update, context, args, is_note):
+        user = update.effective_user
+        chat = update.effective_chat
+        admin_chat = _resolve_admin_group_id(Update)
+
+        if admin_chat.type != "group":  # group only command
+            context.bot.send_message(chat_id=chat, text="This command can only be used for groups.")
+            return
+        
+        if args.bad_user:
+            try:
+                bad_user = client.updater.bot.get_chat_member(chat_id=admin_chat.id, user_id=args.bad_user).user  # type: ignore
+                if args.warn_message:
+                    reason = args.warn_message
+                else:
+                    reason = "No reason provided."
+            except telegram.error.BadRequest:
+                raise exceptions.UserNotFoundException(args.bad_user)
+                return
+        else: 
+            context.bot.send_message(chat_id=chat.id, text="Invalid arguments. Please provide a user ID/mention and a reason.")
+            return
+
+        try:
+            warning_entry = models.Warning(
+                user_id=bad_user.id,
+                date_added=datetime.datetime.now(timezone.utc),
+                warned_by=user.username,
+                warned_by_id=user.id,
+                reason=reason
+            )
+
+            if is_note:
+                warning_entry.is_usernote = True
```

Comment:
Why not move `is_usernote=is_note` into the instantiation?

```suggestion
            warning_entry = models.Warning(
                user_id=bad_user.id,
                date_added=datetime.datetime.now(timezone.utc),
                warned_by=user.username,
                warned_by_id=user.id,
                reason=reason,
                is_usernote=is_note,
            )
```